tests/linalg_test.py

```diff
@@ -577,6 +577,22 @@ def args_maker():
                             check_dtypes=True, tol=1e-3)
     self._CompileAndCheck(np.linalg.inv, args_maker, check_dtypes=True)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name":
+       "_shape={}".format(jtu.format_shape_dtype_string(shape, dtype)),
+       "shape": shape, "dtype": dtype, "rng_factory": rng_factory}
+      for shape in [(1, 1), (4, 4), (2, 70, 7), (2e4, 7), (7, 1e5), (70, 7, 2)]
```

Comment:
Floats aren't valid in NumPy shape parameters.

Try:
```suggestion
      for shape in [(1, 1), (4, 4), (2, 70, 7), (int(2e4), 7), (7, int(1e5)), (70, 7, 2)]
```
or:
```suggestion
      for shape in [(1, 1), (4, 4), (2, 70, 7), (2000, 7), (7, 10000), (70, 7, 2)]
```

Reply:
thanks for the detailed feedbacks, @shoyer !

Reply:
Tests are failing with a dtype mismatch (results returning `float64` instead of `float32`), but only for `(70,7,2)` case...and only for certain versions of Python. Do you have an idea what could go wrong here ?

Reply:
Ah, I suspect this is related to #768. Would it be okay if I default `onp.linalg.pinv` to return `float64` instead? -- This might instead break things when jax.numpy actually returns `float32`. Let me know what you think!

Reply:
(@Shoyer @TuanNguyen27 Thanks for your incredible work on this so far, and attention to detail!)

Re: dtype issues and #768, those are tricky to get right, but they should always be solvable by explicitly "casting" Python literals to specific dtypes, using things like `np.float32(10)` in place of just the literal `10`. Maybe the literals in the code need to be cast that way to avoid NumPy automatic promotion?

Reply:
Both `np.float32(10)` and `rcond = np.asarray(rcond, dtype=a.dtype)` didn't work, but casting the final result `lax.convert_element_type(res, a.dtype)` seems to do the trick. Is this okay ?

Reply:
@shoyer pinging you again in case this is good to merge !